src/lib/redux/store.ts

```diff
@@ -1,59 +1,41 @@
-import { PreloadedState, combineReducers, configureStore, createAsyncThunk } from "@reduxjs/toolkit";
+import { PreloadedState, configureStore } from "@reduxjs/toolkit";
 import { listenerMiddleware } from "./listenerMiddleware";
-import { createServices } from "../services";
-
-import { appSlice } from "./slices/app";
-import { deviceCredentialsSlice } from "./slices/deviceCredentials";
-import { organizationSlice } from "./slices/organization";
-import { roomSlice } from "./slices/room";
-import { roomConnectionSlice } from "./slices/roomConnection";
-import { signalConnectionSlice } from "./slices/signalConnection";
-import { rtcConnectionSlice } from "./slices/rtcConnection";
-import { localMediaSlice } from "./slices/localMedia";
-
-export const rootReducer = combineReducers({
-    app: appSlice.reducer,
-    deviceCredentials: deviceCredentialsSlice.reducer,
-    localMedia: localMediaSlice.reducer,
-    organization: organizationSlice.reducer,
-    room: roomSlice.reducer,
-    roomConnection: roomConnectionSlice.reducer,
-    rtcConnection: rtcConnectionSlice.reducer,
-    signalConnection: signalConnectionSlice.reducer,
-});
+import { RootState, rootReducer } from "./reducer";
 
 export const createStore = ({
     preloadedState,
-    injectServices,
-    dispatchEvent,
 }: {
     preloadedState?: PreloadedState<RootState>;
-    injectServices: ReturnType<typeof createServices>;
-    dispatchEvent: (event: CustomEvent) => void;
+    // dispatchEvent: (event: CustomEvent) => void;
 }) => {
     return configureStore({
         reducer: rootReducer,
         middleware: (getDefaultMiddleware) =>
             getDefaultMiddleware({
                 thunk: {
-                    extraArgument: { services: injectServices, dispatchEvent },
+                    extraArgument: {},
                 },
                 serializableCheck: false,
             }).prepend(listenerMiddleware.middleware),
         preloadedState,
     });
 };
 
-export type RootState = ReturnType<typeof rootReducer>;
-export type AppDispatch = ReturnType<typeof createStore>["dispatch"];
-export type ThunkConfig = {
-    state: RootState;
-    dispatch: AppDispatch;
-    extra: { services: ReturnType<typeof createServices>; dispatchEvent: (event: CustomEvent) => void };
-};
+export const store = createStore({});
```

Comment:
this does cause one problem...
we can't inject services if we have one `store` instance to be accessed by both hooks, but we can't inject services from `useLocalMedia` because we don't know stuff like the room URL yet

Reply:
we _could_ have a dedicated store for local media, then a roomConnection store would subscribe to it to do stuff like send signal events on mute/unmute etc...

Reply:
This comment might have come before our internal discussion. But I strongly believe LocalMedia should have its own store. Then RoomConnection could subscribe to it directly, or use the events which LocalMedia already dispatches.